discovery/azure/azure.go

```diff
@@ -552,7 +552,7 @@ func (client *azureClient) getNetworkInterfaceByID(networkInterfaceID string) (n
 
 	result, err = client.nic.GetResponder(resp)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
+		return result, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
```

Comment:
I think it's unnecessary.

Reply:
Hmm, can you explain why? This looks like a valid change to make and if nothing, we should atleast document why the error is being ignored.

From what I see, when it errors, the result will be empty.

Reply:
Agreed, looks to me like this is an error that is slipping through the cracks right now. Might be nicer to declare `var err error` at the top like `result` and keep the `err = autorest...` line as is but change the return at the end of the function to `return result, err`?

Reply:
I see the code that invoke this method and this result unused if err is not nil. So i think is it fixes `return result,...` to `return nil, autorest.NewErrorWithError()` will be better?